src/figuresClasses.ts

```diff
@@ -1,11 +1,98 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: Shape = 'triangle',
+  ) {
+    this.a = a;
+    this.b = b;
+    this.c = c;
+    this.color = color;
+    this.shape = 'triangle';
```

Comment:
The constructor explicitly assigns values to class properties which is unnecessary because TypeScript does this automatically when using public in the parameter list. Remove the manual assignments to `a`, `b`, `c`, `color`, and `shape`.